julia/src/evaluate.jl

```diff
@@ -13,9 +13,37 @@ end
 
 Base.show(io::IO, atom::Atom) = show(io, print(atom.ptr))
 
+"""
+    juliaize(x)
+
+Get the corresponding Julia object from some output of Dex.
+"""
+juliaize(x::CAtom) = bust_union(x)
+juliaize(x::Ptr{HsAtom}) = juliaize(CAtom(x))
 juliaize(x::Atom) = juliaize(x.ptr)
 Base.convert(::Type{T}, atom::Atom) where {T<:Number} = convert(T, juliaize(atom))
 
+"""
+    dexize(x)
+
+Get the corresponding Dex object from some output of Julia.
+
+NB: this is currently a hack that goes via string processing.
+"""
+function dexize(x::Float32, _module=PRELUDE, env=_module)
```

Comment:
I suspect we do not need to take a `_module` argument

Reply:
The `_module` and `env` parameters are a bit of a hack that I added for Python bindings. The rough idea was that `module` is the module an output atom declares itself to be defined in, while `env` is the scope that's really used to evaluate the expression. This is used in the `__call__` implementation where we temporarily extend the prelude with new names that refer to arguments, but then we want to pretend that the result is still defined in the original module that doesn't have those dummies. But now that I think about it, it's only well defined for non-dependent functions, so we should find a different workaround...

Reply:
In any case, I think you can safely ignore `env` and just use the `_module`.